libs/cart/src/drivers/magento/cart-billing-address.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { Apollo } from 'apollo-angular';
+import { Observable, zip } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { DaffCartBillingAddressServiceInterface } from '../interfaces/cart-billing-address-service.interface';
+import { DaffCart } from '../../models/cart';
+import { DaffCartAddress } from '../../models/cart-address';
+import {
+  getBillingAddress,
+  updateBillingAddress,
+  setGuestEmail
+} from './queries';
+import {
+  MagentoGetBillingAddressResponse,
+  MagentoUpdateBillingAddressResponse,
+  MagentoSetGuestEmailResponse
+} from './models/responses';
+import { DaffMagentoBillingAddressInputTransformer } from './transforms/inputs/billing-address.service';
+import { DaffMagentoCartTransformer } from './transforms/outputs/cart.service';
+import { DaffMagentoBillingAddressTransformer } from './transforms/outputs/billing-address.service';
+
+/**
+ * A service for making Magento GraphQL queries for carts.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoCartBillingAddressService implements DaffCartBillingAddressServiceInterface<DaffCartAddress, DaffCart> {
+  constructor(
+    private apollo: Apollo,
+    public cartTransformer: DaffMagentoCartTransformer,
+    public billingAddressTransformer: DaffMagentoBillingAddressTransformer,
+    public billingAddressInputTransformer: DaffMagentoBillingAddressInputTransformer
+  ) {}
+
+  get(cartId: string): Observable<DaffCartAddress> {
+    return this.apollo.query<MagentoGetBillingAddressResponse>({
+      query: getBillingAddress,
+      variables: {cartId}
+    }).pipe(
+      map(result => this.billingAddressTransformer.transform({
+        ...result.data.cart.billing_address,
+        email: result.data.cart.email
+      }))
+    )
+  }
+
+  update(cartId: string, address: Partial<DaffCartAddress>): Observable<Partial<DaffCart>> {
+    return zip(
+      this.apollo.mutate<MagentoUpdateBillingAddressResponse>({
+        mutation: updateBillingAddress,
+        variables: {
+          cartId,
+          address: this.billingAddressInputTransformer.transform(address)
+        }
+      }),
+      this.apollo.mutate<MagentoSetGuestEmailResponse>({
+        mutation: setGuestEmail,
+        variables: {
+          cartId,
+          email: address.email
+        }
+      })
+    ).pipe(
+      map(([updateResult, setEmailResult]) => this.cartTransformer.transform({
+        ...updateResult.data.setBillingAddressOnCart.cart,
+        // only set the email from the set email result since we don't know if the billing address update is included in the response
```

Comment:
Do we not know this, because we're not sure which mutation will finish first? Like if the setGuestEmail mutation finishes, then the updateBillingAddress returns after, the email might be on the updateBillingAddress response?

Reply:
Correct except we don't really care if the billing address response has the updated email or not since we are explicitly setting it from the set email response. I added the comment to explain why I wasn't spreading `setEmailResult` in, i.e. we don't know if the rest of `setEmailResult` is the most up to date.

This is based on the assumption that setting the email on the cart only changes the email and nothing else. Updating the billing address is more likely to affect other parts of the cart like `payment` for example.

Reply:
With the recent change to the set email mutation, this comment is probably not needed.